### ISSUE TITLE
material.Material class internal updates still broken

### ISSUE THREAD

**Issue by joelvbernier** [HEXRD/hexrd#217]
The following code snippet illustrates the issue.  Fundamentally, if I change the `sgnum` on the default material (say to quickly make an HCP material), then update the `latticeParameters` object, the `planeData` object is not getting updated.  However, if I re-assign the `sgnum`, it looks to be properly updated.  Seems that the issue is that updating `Material.latticeParameters` is not triggering an update of `Material.planeData.lparms`, which leads to the incosistency:
```
import numpy as np

from hexrd import material
from hexrd import valunits

matl = material.Material(
    "Titanium",
    dmin=valunits.valWUnit("dmin", "LENGTH", 0.35, "angstrom")
)

# switch to HCP again...
matl.sgnum = 194

matl.latticeParameters[0] = valunits.valWUnit(
    "a", "LENGTH", 2.9508, "angstrom")
matl.latticeParameters[1] = valunits.valWUnit(
    "b", "LENGTH", 2.9508, "angstrom")
matl.latticeParameters[2] = valunits.valWUnit(
    "c", "LENGTH", 4.6855, "angstrom")

print("material says lattice parameters are:")
print(matl.latticeParameters)
print("matl.planeData says:")
print(matl.planeData.lparms)
print("these are wrong!")
print(
    np.vstack([matl.planeData.hkls, matl.planeData.getPlaneSpacings()]).T
)

# switch to HCP
matl.sgnum = 194

print("material says lattice parameters are:")
print(matl.latticeParameters)
print("matl.planeData says:")
print(matl.planeData.lparms)
print("these seem correct now:")
print(
    np.vstack([matl.planeData.hkls, matl.planeData.getPlaneSpacings()]).T
)
```

**Comment by psavery** [comment]
@saransh13 Did this get fixed by #218?

**Comment by saransh13** [comment]
Yep, I believe so. We can close the issue.
